pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -364,6 +393,9 @@ type PipelineSpec struct {
 	// +kubebuilder:default={"disabled": false}
 	// +optional
 	Watermark Watermark `json:"watermark,omitempty" protobuf:"bytes,6,opt,name=watermark"`
+	// Daemon is used to customize the Daemon Deployment
+	// +optional
+	Daemon Daemon `json:"daemon,omitempty" protobuf:"bytes,7,rep,name=daemon"`
```

Comment:
thinking whether we nest it under `LifeCycle`, this is because we have to also add Job resources and those jobs are for setting up and tearing down the pipeline. The Daemon Server is also used to manage the lifecycle too, e.g., pausing the pipeline, etc.